# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsDSURIContentListener.h
# Commit: baffa8a7b961
# Full Hash: baffa8a7b9618e9a02e4bd808adb71dead99cce5
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-11-12 09:43:07
# Regressor Bug: 1589270
# File Overlap Count: 2
# Description:
#   Bug 1589270 - Part 3: Convert nsExternalHelperApp to use BrowsingContext instead of nsIInterfaceRequestor. r=bzbarsky
#   
#   This also converts MaybeCloseWindowHelper, and results in the window close operations being always run in the parent (even without DocumentChannel).
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49528
# ==============================================================================

diff -r 68bc2ae9b89a -r baffa8a7b961 docshell/base/nsDSURIContentListener.h
--- a/docshell/base/nsDSURIContentListener.h	Fri Nov 08 03:01:03 2019 +0000
+++ b/docshell/base/nsDSURIContentListener.h	Fri Nov 08 04:35:05 2019 +0000
@@ -23,13 +23,15 @@
   NS_DECL_ISUPPORTS
   NS_DECL_NSITIMERCALLBACK
 
-  explicit MaybeCloseWindowHelper(nsIInterfaceRequestor* aContentContext);
+  explicit MaybeCloseWindowHelper(
+      mozilla::dom::BrowsingContext* aContentContext);
 
   /**
    * Closes the provided window async (if mShouldCloseWindow is true)
-   * and returns its opener if the window was just openend.
+   * and returns its opener if the window was just opened. Otherwise
+   * returns the BrowsingContext provided in the constructor.
    */
-  nsIInterfaceRequestor* MaybeCloseWindow();
+  mozilla::dom::BrowsingContext* MaybeCloseWindow();
 
   void SetShouldCloseWindow(bool aShouldCloseWindow);
 
@@ -40,13 +42,13 @@
   /**
    * The dom window associated to handle content.
    */
-  nsCOMPtr<nsIInterfaceRequestor> mContentContext;
+  RefPtr<mozilla::dom::BrowsingContext> mBrowsingContext;
 
   /**
    * Used to close the window on a timer, to avoid any exceptions that are
    * thrown if we try to close the window before it's fully loaded.
    */
-  nsCOMPtr<nsPIDOMWindowOuter> mWindowToClose;
+  RefPtr<mozilla::dom::BrowsingContext> mBCToClose;
   nsCOMPtr<nsITimer> mTimer;
 
   /**